website/content/en/preview/getting-started/_index.md

```diff
@@ -13,3 +13,211 @@ Learn more about Karpenter and how to get started below.
+
+If you have not already configured Karpenter when you set up your Kubernetes cluster in AWS, follow the instructions below to add Karpenter to your existing cluster.
+
+## Add Karpenter to an existing cluster
+
+To add Karpenter to your cluster, you need to create IAM roles, add tags to subnets and security groups, and update the aws-auth ConfigMap.
+
+### Create IAM role
+
+To get started, create a new IAM role for the Karpenter controller.
+
+The controller will be using [IAM Roles for Service Accounts (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) which requires an OIDC endpoint.
+
+If you have another option for using IAM credentials with workloads (e.g. [kube2iam](https://github.com/jtblin/kube2iam)) your steps will be different.
+
+Set a variable for your cluster name.
+
+```bash
+CLUSTER_NAME=<your cluster name>
+```
+
+Set other variables from your cluster configuration.
+
+{{% script file="./scripts/step04-env.sh" language="bash" %}}
+
+Use that information to create our IAM role, inline policy, and trust relationship.
+
+{{% script file="./scripts/step05-controller-iam.sh" language="bash" %}}
+
+### Add tags to subnets and security groups
+
+Next add tags to your nodegroup subnets so Karpenter will know which subnets to use.
+
+{{% script file="./scripts/step06-tag-subnets.sh" language="bash" %}}
+
+Add tags to the security groups.
+This command only tags the security groups for the first nodegroup in the cluster.
+If you have multiple nodegroups or multiple security groups you will need to decide which one Karpenter should use.
+
+{{% script file="./scripts/step07-tag-security-groups.sh" language="bash" %}}
+
+### Deploy Karpenter
+
+First set the Karpenter release you want to deploy and the IAM instance profile that is used by the existing nodes in your cluster.
+```bash
+export KARPENTER_VERSION={{< param "latest_release_version" >}}
+export IAM_INSTANCE_PROFILE_NAME=<your instance profile name>
+```
+
+We can now generate a full Karpenter deployment yaml from the helm chart.
+
+{{% script file="./scripts/step09-generate-chart.sh" language="bash" %}}
+
+Modify the following lines in the karpenter.yaml file.
+
+### Set node affinity
+
+Edit the karpenter.yaml file and find the karpenter deployment affinity rules.
+Modify the affinity so karpenter will run on one of the existing node group nodes.
+
+The rules should look something like this.
+Modify the value to match your `$NODEGROUP`.
+
+```
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: karpenter.sh/provisioner-name
+                operator: DoesNotExist
+            - matchExpressions:
+              - key: eks.amazonaws.com/nodegroup
+                operator: In
+                values:
+                - ${NODEGROUP}
```

Comment:
I wonder if this is necessary? It should be enough to simply say "don't run karpenter on itself", which is already included in the helm chart.

Reply:
IIUC, this is from the CAS migration guide so that the user can install Karpenter on a nodegroup that the user isn't planning on deleting.

Reply:
Yes, this was taken from the CAS migration guide. Should I change it?